### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

**Review by voloshynamaryna11 — COMMENTED** [review]
everything is perfect

File: src/main/java/com/internet/shop/controllers/security/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("pwd");
+        try {
+            User user = authService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/security/LoginController.java]: let's make userId a constant

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -46,4 +46,10 @@ public boolean deleteById(Long id) {
     public boolean delete(User userDelete) {
         return Storage.users.removeIf(user -> (user.getId().equals(userDelete.getId())));
     }
+
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream()
+                .filter(s -> s.getLogin().equals(login))
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]:
```suggestion
                .filter(u -> u.getLogin().equals(login))
```

File: src/main/java/com/internet/shop/security/AuthenticationImpl.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDb = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect login or password."));
+        if (userFromDb.getPassword().equals(password)) {
+            return userFromDb;
+        }
+        throw new AuthenticationException("Incorrect login or password.");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationImpl.java]: either make both checks in one if() or make exception message a constant

File: src/main/java/com/internet/shop/security/AuthenticationService.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.internet.shop.security;
+
+import com.internet.shop.exceptions.AuthenticationException;
+import com.internet.shop.model.User;
+
+public interface AuthenticationService {
+    User login(String login, String password) throws AuthenticationException;
+}
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationService.java]:
```suggestion
```

File: src/main/java/com/internet/shop/service/UserService.java [hunk 5]
```diff
@@ -13,4 +14,7 @@
     User update(User user);
 
     boolean delete(Long id);
+
+    Optional<User> findByLogin(String login);
 }
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/service/UserService.java]:
```suggestion
```
at the end of file should be only one empty line

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    public static final Injector injector = Injector.getInstance("com.internet.shop");
+    private static final String USER_ID = "user_id";
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        if (userId == null || userService.get(userId) == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java]: done

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java (hunk 6 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: done

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java (hunk 7 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: done

File: src/main/java/com/internet/shop/service/UserService.java (hunk 5 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/service/UserService.java]: done

File: src/main/java/com/internet/shop/security/AuthenticationService.java (hunk 4 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/security/AuthenticationService.java]: done

File: src/main/java/com/internet/shop/controllers/security/LoginController.java (hunk 1 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/controllers/security/LoginController.java]: done

File: src/main/java/com/internet/shop/security/AuthenticationImpl.java (hunk 3 above)
Comment (IrynaDim) [reply src/main/java/com/internet/shop/security/AuthenticationImpl.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**